#!/usr/bin/env python
PACKAGE = "mrs_controllers"

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kiwxy", double_t, 0, "Integral constant for xy-axes", 0.0, 0.0, 10.0)
gen.add("kibxy", double_t, 1, "Integral constant for xy-axes", 0.0, 0.0, 10.0)
gen.add("kqxy", double_t, 2, "Attitude constant for xy-axis", 0.0, 0.0, 20.0)
gen.add("kqz", double_t, 4, "Attitude constant for z-axis", 0.0, 0.0, 40.0)
gen.add("kiwxy_lim", double_t, 8, "xy-axes integral limit", 0.0, 0.0, 10.0)
gen.add("kibxy_lim", double_t, 16, "xy-axes integral limit", 0.0, 0.0, 10.0)
gen.add("km", double_t, 32, "Integral constant for mass", 0.0, 0.0, 2.0)
gen.add("km_lim", double_t, 64, "mass integral limit", 0.0, 0.0, 10.0)

exit(gen.generate(PACKAGE, "MpcController", "mpc_controller"))
